Makefile

```diff
@@ -184,53 +177,27 @@ endif
 generate-cov: ## Generate coverage report for Cilium integration-tests.
 	# Remove generated code from coverage
 	$(QUIET) grep -Ev '(^github.com/cilium/cilium/api/v1)|(generated.deepcopy.go)|(^github.com/cilium/cilium/pkg/k8s/client/)' \
-		coverage-all-tmp.out > coverage-all.out
-	$(QUIET)$(GO) tool cover -html=coverage-all.out -o=coverage-all.html
-	$(QUIET) rm coverage.out coverage-all-tmp.out
+		coverage.out > coverage.out.tmp
+	$(QUIET)$(GO) tool cover -html=coverage.out.tmp -o=coverage-all.html
+	$(QUIET) rm coverage.out.tmp
 	@rmdir ./daemon/1 ./daemon/1_backup 2> /dev/null || true
 
-init-coverage: ## Initialize converage report for Cilium integration-tests.
-	$(QUIET) echo "mode: count" > coverage-all-tmp.out
-	$(QUIET) echo "mode: count" > coverage.out
-
 integration-tests: GO_TAGS_FLAGS+=integration_tests
 integration-tests: start-kvstores ## Runs all integration-tests for Cilium.
 	$(QUIET) $(MAKE) $(SUBMAKEOPTS) -C test/bpf/
 ifeq ($(SKIP_VET),"false")
 	$(MAKE) govet
 endif
-	$(MAKE) init-coverage
-	# It seems that in some env if the path is large enough for the full list
-	# of files, the full bash command in that target gets too big for bash and
-	# hence will trigger an error of too many arguments. As a workaround, we
-	# have to process these packages in different subshells.
-	for pkg in $(TESTPKGS); do \
-		>&2 $(ECHO_TEST) $$pkg; \
-		$(GO_TEST) $(TEST_UNITTEST_LDFLAGS) $$pkg $(GOTEST_BASE) $(GOTEST_COVER_OPTS) -coverpkg $$pkg \
-		|| exit 1; \
-		tail -n +2 coverage.out >> coverage-all-tmp.out; \
-	done | $(GOTEST_FORMATTER)
+	$(GO_TEST) $(TEST_UNITTEST_LDFLAGS) $(TESTPKGS) $(GOTEST_BASE) $(GOTEST_COVER_OPTS) | $(GOTEST_FORMATTER)
```

Comment:
If the error still exists today, maybe `xargs` could help?

```
$ seq 1 20 | xargs -s 16 echo
1 2 3 4 5
6 7 8 9 10
11 12 13
14 15 16
17 18 19
20
```

```
-s max-chars, --max-chars=max-chars
    Use at most max-chars characters per command line, including the command and ini‐
    tial-arguments and the terminating nulls at the ends  of  the  argument  strings.
    The  largest allowed value is system-dependent, and is calculated as the argument
    length limit for exec, less the size of your  environment,  less  2048  bytes  of
    headroom.   If this value is more than 128KiB, 128Kib is used as the default val‐
    ue; otherwise, the default value is the maximum.  1KiB is 1024 bytes.  xargs  au‐
    tomatically adapts to tighter constraints.
```

(Reading the above, it defaults to the max value so should be able to split into multiple command lines as required?)

Reply:
Thinking further, the strategy above works with `seq` because it generates output on stdin. The problem is that the output of `go list ./...` is slurped into a `make` variable, and getting the variable's contents back out is the problem. I haven't found a way for to get a Make variable's contents written to a file or presented to a shell's stdin. A common pattern would be `echo "$(MAKE_VARIABLE)" | xargs ...`, but the size of the argument to `echo` is the problem.

Reply:
> but the size of the argument to echo is the problem.

I figured it might, but wasn't sure. OK, thanks for the details

> I haven't found a way for to get a Make variable's contents written to a file or presented to a shell's stdin

Here-doc (`<<`) with Makefile variable in the middle? :upside_down_face: (not tested though)

Reply:
Looks like `$(file ...)` can do it ([doc](https://www.gnu.org/software/make/manual/html_node/File-Function.html)):

```Makefile
AAA:=$(shell python3 -c 'print("A"*3000000)')

lots_of_As.txt:
        $(file >$@,$(AAA))
        wc -c $@
```

```
$ make
wc -c lots_of_As.txt
3000001 lots_of_As.txt
```

Reply:
Interesting, didn't know about `file`. Looks like that approach requires a tempfile, it can't pass the contents of a variable directly to `xargs` stdin?

Anyway, the point of this PR is to just be able to use `./...` to avoid Makefile voodoo.

Reply:
> Looks like that approach requires a tempfile, it can't pass the contents of a variable directly to xargs stdin?

I guess not, at least I don't know a way to do so. You'd probably need a temp file indeed (then `xargs -a` for reading from it). But yeah, not required here anyway :)